From a1fa4b1c64755e4171eee5481cab041937742148 Mon Sep 17 00:00:00 2001
From: Marek Marczykowski <marmarek@invisiblethingslab.com>
Date: Fri, 5 Apr 2013 01:41:04 +0200
Subject: [PATCH] libxl: use disk 'script' attribute
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Implement handling of previously introduced <script/> element for disk
config.
This can be used for custom backend configuration like non-standard
device-mapper nodes, or to prepare device in other domain (see the next
patch).

Signed-off-by: Marek Marczykowski-Górecki <marmarek@invisiblethingslab.com>
---
 src/libxl/libxl_conf.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/libxl/libxl_conf.c b/src/libxl/libxl_conf.c
index 3bb5c1e591..8bebc82c2f 100644
--- a/src/libxl/libxl_conf.c
+++ b/src/libxl/libxl_conf.c
@@ -1181,6 +1181,8 @@ libxlMakeDisk(virDomainDiskDef *l_disk, libxl_device_disk *x_disk)
 
     x_disk->vdev = g_strdup(l_disk->dst);
 
+    x_disk->script = g_strdup(l_disk->script);
+
     if (driver) {
         if (STREQ(driver, "tap") || STREQ(driver, "tap2")) {
             switch (format) {
-- 
2.54.0

